docs/deploy/Deploy-Infrastructure.bicep

```diff
@@ -54,14 +65,48 @@ resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-03
   })
 }
 
-// Deploy application insights for collection of application logs and metrics
-module applicationInsightsDeploy 'Deploy-AppInsights.bicep' = if (deployApplicationInsights) {
-  name: 'applicationInsightsDeploy'
-  scope: resourceGroup()
-  params: {
-    location: location
-    envName: envName
-    logAnalyticsWorkspaceId: logAnalyticsWorkspace.id
+// Deploy application insights for receiving azure monitor telemetry
+var applicationInsightsName = '${envName}-ai'
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' = if (deployApplicationInsights) {
+  name: applicationInsightsName
+  location: location
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+    WorkspaceResourceId: logAnalyticsWorkspace.id
+    DisableLocalAuth: true
+  }
+}
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-04-01-preview' existing = if (deployApplicationInsights) {
+  name: keyVaultName
+}
+
+var applicationInsightsConnectionStringSecretName = '${applicationInsightsName}-conn-string'
+resource applicationInsightsConnectionStringSecret 'Microsoft.KeyVault/vaults/secrets@2021-04-01-preview' = if (deployApplicationInsights) {
+  parent: keyVault
+  name: applicationInsightsConnectionStringSecretName
+  properties: {
+    value: deployApplicationInsights ? applicationInsights.properties.ConnectionString : 'default'

```

Comment:
Isn't this check redundant here if we are already checking when deploying this resource?

Reply:
I thought so, but it seems that the value is still evaluated even when we don't deploy the resource, so it was causing a resource not found error in cases where the bool was false. Adding this check resolved the issue.